core/autocrypt/account.py

```diff
@@ -288,41 +194,52 @@ def process_incoming(self, msg, delivto=None):
 
         :type msg: email.message.Message
         :param msg: instance of a standard email Message.
-        :rtype: PeerInfo
+        :rtype: PeerState
         """
         if delivto is None:
             _, delivto = mime.parse_email_addr(msg.get("Delivered-To"))
             assert delivto
-        ident = self.get_identity_from_emailadr([delivto])
+        ident = self.get_identity_from_emailadr(delivto)
         if ident is None:
             raise IdentityNotFound("no identity matches emails={}".format([delivto]))
         From = mime.parse_email_addr(msg["From"])[1]
-        old = ident.config.peers.get(From, {})
+        peerstate = ident.get_peerstate(From)
+        peerchain = peerstate.peerchain
```

Comment:
yes, i think it makes sense to put "modification" operations on the state objects as well.  note that with Kappa they call it "view" rather than state. I choose the AC-level term for now.   In any case, not having account keep direct references to chains makes sense.